schema.json

```diff
@@ -0,0 +1,214 @@
+{
+  "id": "http://api.codeclimate.com/test-coverage-schema#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+
+  "title": "Test Coverage",
+  "description": "Schema that describes the format of the test coverage payload sent to Code Climate",
+
+  "definitions": {
+    "covered_percent": {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 100,
+    },
+    "covered_strength": {
+      "type": "integer",
+      "minimum": 0,
+    },
+
+    "line_counts": {
+      "type": "object",
+      "properties": {
+        "covered": {
+          "description": "Number of covered lines of code",
+          "type": "integer",
+          "minimum": 0,
+        },
+        "missed": {
+          "description": "Number of uncovered lines of code",
+          "type": "integer",
+          "minimum": 0,
+        },
+        "total": {
+          "description": "Total lines of code",
+          "type": "integer",
+          "minimum": 0,
+        },
+      },
+      "required": ["covered", "missed", "total"],
+    },
+
+    "git_sha": {
+      "type": "string",
+      "pattern": "^\d{1,}\.\d{1,}\.\d{1,}",
```

Comment:
This looks like a version string regex, not a SHA regex. If we're enforcing the SHA format, we need to be cognizant of the if/when git moves away from SHA-1.